packages/core/saved-objects/core-saved-objects-migration-server-internal/src/model/model.ts

```diff
@@ -1474,9 +1488,19 @@ export const model = (currentState: State, resW: ResponseType<AllActionStates>):
   } else if (stateP.controlState === 'CREATE_NEW_TARGET') {
     const res = resW as ExcludeRetryableEsError<ResponseType<typeof stateP.controlState>>;
     if (Either.isRight(res)) {
+      if (res.right === 'index_already_exists') {
```

Comment:
This the second fix of the PR:
If we are creating a new index but the index already exists:

- it probably belongs to a previous failed upgrade
  - which managed to create the index
  - but failed before it could the aliases

In this scenario, instead of simply completing the migration, we will attempt to update mappings first.

Reply:
I was wondering why `index_already_exists` was a `right` response. 
Strictly speaking, if the index exists, and it shouldn't, then the previous migration failure would be an error state, i.e. a `left` response). But I see what we're doing here, migrations skip over index creation and move on. Hence, it's a recoverable 'unexpected' case. 
It makes sense now, great!